File: ignite-spring/src/main/kotlin/org/rookedsysc/ignitespring/comment/mapper/CommentMapper.kt
package org.rookedsysc.ignitespring.comment.mapper

import org.rookedsysc.ignitespring.comment.entity.CommentEntity
import org.rookedsysc.ignitespring.comment.infrastructure.web.dto.CommentCreateDto
import org.rookedsysc.ignitespring.comment.infrastructure.web.dto.CommentListDto
import org.rookedsysc.ignitespring.comment.infrastructure.web.dto.CommentPostDetailListDto
import org.rookedsysc.ignitespring.post.infrastructure.web.dto.PostListDto

class CommentMapper {
    companion object {
        fun toEntity(postId: Long, commentCreateDto: CommentCreateDto): CommentEntity {
            return CommentEntity(
                postId = postId,
                content = commentCreateDto.content,
                title = commentCreateDto.title
            )
        }

        fun toCommentListDto(commentEntity: CommentEntity, postListDto: PostListDto): CommentListDto {
            return CommentListDto(
                id = commentEntity.id,
                title = commentEntity.title,
                content = commentEntity.content,
                createdAt = commentEntity.createdAt,
                postListDto = postListDto
            )
        }

        fun toCommentPostDetailListDto(comment: CommentEntity): CommentPostDetailListDto {
            return CommentPostDetailListDto(
                id = comment.id,
                title = comment.title,
                content = comment.content,
                createdAt = comment.createdAt
            )
        }
    }
}

File: ignite-nest/src/comment/mapper/comment.mapper.ts
import { CommentUploadDto } from '../infrastructure/web/dto/request/comment-upload.dto';
import { Comment } from '../entity/comment.entity';
import { Post } from '../../post/entity/post.entity';
import { PostListDto } from '../../post/infrastructure/web/dto/response/post-list.dto';
import { CommentListDto } from '../infrastructure/web/dto/response/comment-list.dto';

export class CommentMapper {
  static toEntity(post: Post, comment: CommentUploadDto): Comment {
    return new Comment(post, comment.content, comment.content);
  }

  static toResponse(
    comment: Comment,
    postListDto: PostListDto,
  ): CommentListDto {
    const commentListDto = {
      id: comment.id,
      title: comment.title,
      content: comment.content,
      createdAt: comment.createdAt,
      postListDto,
    };
    return commentListDto;
  }
}
